crates/rspack_plugin_javascript/src/visitors/dependency/hmr_scanner.rs

```diff
@@ -0,0 +1,112 @@
+use rspack_core::ModuleDependency;
+use swc_core::common::pass::AstNodePath;
+use swc_core::ecma::ast::{CallExpr, Expr, Lit, Str};
+use swc_core::ecma::visit::{AstParentNodeRef, VisitAstPath, VisitWithPath};
+
+use super::{
+  as_parent_path, is_import_meta_hot_accept_call, is_import_meta_hot_decline_call,
+  is_module_hot_accept_call, is_module_hot_decline_call,
+};
+use crate::dependency::{
+  ImportMetaModuleHotAcceptDependency, ImportMetaModuleHotDeclineDependency,
+  ModuleHotAcceptDependency, ModuleHotDeclineDependency,
+};
+
+pub struct HmrDependencyScanner<'a> {
+  pub dependencies: &'a mut Vec<Box<dyn ModuleDependency>>,
+  pub flag: (bool, bool, bool, bool),
```

Comment:
Avoid using tuples when a length is greater than 3, It is hard to understand.
I prefer using `Bitflag` , which could reduce another 24 bytes of memory